executor/status_handler.go

```diff
@@ -16,87 +15,89 @@ func (executor *Executor) podsStatusHandler(entityName string, namespace string,
 	msg = "pods restarting exceeded timout (15 min)"
 	start := time.Now()
 
-	for time.Now().Sub(start) < automationsExecutionTimeout {
+	objectName := ""
+	result = proto.AutomationFailed
+	targetPods = 0
+	var err error = nil
+	flag := false
 
-		objectName := ""
-		result = proto.AutomationFailed
-		targetPods = 0
-		var err error = nil
-		flag := false
+	if strings.ToLower(kind) == "statefulset" {
+		objectName, targetPods, err = executor.statefulsetsHandler(entityName, namespace)
+		if err != nil {
+			flag = true
 
-		time.Sleep(podStatusSleep)
+		}
 
-		if strings.ToLower(kind) == "deployment" {
-			objectName, targetPods, err = executor.deploymentsHandler(entityName, namespace)
-			if err != nil {
-				flag = true
+	} else if strings.ToLower(kind) == "daemonset" {
+		objectName, targetPods, err = executor.daemonsetsHandler(entityName, namespace)
+		if err != nil {
+			flag = true
 
-			}
+		}
 
-		} else if strings.ToLower(kind) == "statefulset" {
-			objectName, targetPods, err = executor.statefulsetsHandler(entityName, namespace)
-			if err != nil {
-				flag = true
+	} else if strings.ToLower(kind) == "job" || strings.ToLower(kind) == "cronjob" {
+		job, err := executor.kube.GetCronJob(namespace, entityName)
 
-			}
+		if err != nil {
+			flag = true
 
-		} else if strings.ToLower(kind) == "daemonset" {
-			objectName, targetPods, err = executor.daemonsetsHandler(entityName, namespace)
-			if err != nil {
-				flag = true
+		} else {
+			// get the new job
+			objectName = job.Name
+			targetPods = 1
 
-			}
+		}
+	}
 
-		} else if strings.ToLower(kind) == "job" || strings.ToLower(kind) == "cronjob" {
-			job, err := executor.kube.GetCronJob(namespace, entityName)
+	if flag {
+		msg = "failed to trigger pod status"
+		result = proto.AutomationFailed
 
-			if err != nil {
-				flag = true
+	} else {
+		for time.Now().Sub(start) < automationsExecutionTimeout {
 
-			} else {
-				// get the new job
-				objectName = job.Name
-				targetPods = 1
+			time.Sleep(podStatusSleep)
```

Comment:
put sleep after calling the deployment handler

Reply:
In that case we will check the replica set that is 15 seconds, for me it is better to check it as soon as we get it.

Reply:
as we wait 15 sec anyway it is ok for me aslo